features/upgrade/asb/upgrade.feature

```diff
@@ -0,0 +1,77 @@
+    # When I process and create:
+    #   | f | <%= ENV['BUSHSLICER_HOME'] %>/testdata/olm/art-secret-ansible.yaml |
+    #   | p | NAME=aosqe4-secret                                                                              |
+    #   | p | TOKEN=<your quay token>                                                                         |
+    # Then the step should succeed
+    # When I process and create:
+    #   | f | <%= ENV['BUSHSLICER_HOME'] %>/testdata/olm/operatorsource-ansible.yaml |
+    #   | p | NAME=aosqe4-operators                                                                               |
+    #   | p | SECRET=aosqe4-secret                                                                                |
+    #   | p | REGISTRY=aosqe4                                                                                     |
+    # Then the step should succeed
+    # Enable Service Catalog so that the ASB can work well
+    Given enable service catalog
+    # Get the cluster version: oc get clusterversion version -o=jsonpath='{.spec.channel}' | tr -d 'stable-'
+    Given the major.minor version of the cluster is stored in the clipboard
+
+    # This step will install the OperatorGroup, operator. Please use your operator Package name in here.
+    Given optional operator "openshiftansibleservicebroker" from channel "<%= cb.operator_channel_name %>" is subscribed in "openshift-ansible-service-broker" project
+    # Check the pods of the operator
+    Then I wait for the "openshift-ansible-service-broker-operator" deployment to appear
+    And a pod becomes ready with labels:
+      | name=openshift-ansible-service-broker-operator |
+    # Check if the corresponding CRD is ready
+    Then I check that there are no automationbrokers
+    #Create Cluster Role Binding for the operator
+    When I process and create:
+      | f | <%= ENV['BushiSlicer::HOME'] %>/testdata/olm/asb-rolebinding-template.yaml |
+    Then the step should succeed
+    # Create customer resource for the operator
+    When I process and create:
+      | f | <%= ENV['BushiSlicer::HOME'] %>/testdata/olm/asb-cr-template.yaml |
+    Then the step should succeed
+    And a pod becomes ready with labels:
+      | app=ansible-service-broker |
+    
+  @admin
+  @upgrade-checkclusterservicebroker
+  @users=upuser1,upuser2
+  Scenario: upgrade ASB
+    Given I switch to cluster admin pseudo user
+    And I use the "openshift-ansible-service-broker" project
+    # Check if the previous ASB works works well.
+    Given I wait for the "ansible-service-broker" cluster_service_broker to become ready
+    # Update ASB operator to the new version
+    Given the major.minor version of the cluster is stored in the clipboard
+    Given I successfully merge patch resource "subscription/openshiftansibleservicebroker" with:
+      | {"spec":{"channel":"<%= cb.operator_channel_name %>"}} |
+    And I wait for all clusterserviceversion in the project to become ready up to 360 seconds
```

Comment:
@akostadinov, based on comments and change it this way encounters 
```
      BushSlicer::ClusterServiceVersion openshiftansibleservicebroker.4.3.12-202004091734 did not become ready within timeout (RuntimeError)
      /home/chuo/verification-tests/features/step_definitions/resources.rb:107:in `block (2 levels) in <top (required)>'
      /home/chuo/verification-tests/features/step_definitions/resources.rb:103:in `map'
      /home/chuo/verification-tests/features/step_definitions/resources.rb:103:in `/^(I|admin) waits? for all (\w+) in the(?: "(.+?)")? project to become ready(?: up to (\d+) seconds)?$/'
      features/upgrade/asb/upgrade.feature:65:in `And I wait for all clusterserviceversion in the project to become ready up to 360 seconds'

```
also have ready? defined in lib/openshift/cluster_service_version.rb. Could you help to have a look?

Reply:
> @akostadinov @jianzhangbjz , based on comments and change it this way encounters
> 
> ```
>       BushSlicer::ClusterServiceVersion openshiftansibleservicebroker.4.3.12-202004091734 did not become ready within timeout (RuntimeError)
>       /home/chuo/verification-tests/features/step_definitions/resources.rb:107:in `block (2 levels) in <top (required)>'
>       /home/chuo/verification-tests/features/step_definitions/resources.rb:103:in `map'
>       /home/chuo/verification-tests/features/step_definitions/resources.rb:103:in `/^(I|admin) waits? for all (\w+) in the(?: "(.+?)")? project to become ready(?: up to (\d+) seconds)?$/'
>       features/upgrade/asb/upgrade.feature:65:in `And I wait for all clusterserviceversion in the project to become ready up to 360 seconds'
> ```
> 
> also have ready? defined in lib/openshift/cluster_service_version.rb. Could you help to have a look?

Resolved.